### ISSUE TITLE
Failures against Content Security Policy of your site blocks the use of 'eval' in JavaScript

### ISSUE THREAD

**Issue by hsdk123** [emscripten-core/emscripten#12910]
Hi, I'm noticing that the .js main application created through emscripten/wasm is blocked by chrome when using 
`"Content-Security-Policy", "script-src 'self'"`
on the server side.

This is a security issue, would love an update on this.

**Comment by SomeDevHere** [comment]
In Chrome when `Content-Security-Policy` is specified but `script-src` does not include `unsafe-eval` WebAssembly does not work as all functions that load wasm code cannot be used.
See this issue for more information: https://github.com/WebAssembly/content-security-policy/issues/7

**Comment by soffieprincess07** [comment]
> In Chrome when `Content-Security-Policy` is specified but `script-src` does not include `unsafe-eval` WebAssembly does not work as all functions that load wasm code cannot be used.
> See this issue for more information: [WebAssembly/content-security-policy#7](https://github.com/WebAssembly/content-security-policy/issues/7)

**Comment by soffieprincess07** [comment]
> Hi, I'm noticing that the .js main application created through emscripten/wasm is blocked by chrome when using
> `"Content-Security-Policy", "script-src 'self'"`
> on the server side.
> 
> This is a security issue, would love an update on this.

**Comment by kripken** [comment]
@fgmccabe do you perhaps have a good link to explain these issues, and what people can do about them?

**Comment by fgmccabe** [comment]
Using 'self' as the policy source blocks javascript eval. It also blocks wasm. 
Simple solution (until wasm-unsafe-eval is released)
"Content-Security-Policy", "script-src 'self' 'unsafe-eval'"

This will allow JavaScript eval though as well.

In the future:

"Content-Security-Policy", "script-src 'self' 'wasm-usafe-eval'"

**Comment by sbc100** [comment]
Is this needed even if we only use streaming compilation/instantiation?  (i.e the code is linked directly back to a URL?)

**Comment by fgmccabe** [comment]
yes. There will be no special handling of streaming for the foreseeable.
